lib/tessel/deploy.js

```diff
@@ -241,7 +242,7 @@ actions.tarBundle = function(opts) {
     });
 
     if (opts.single) {
-      fstream.addIgnoreRules(['*', '!' + opts.entryPoint]);
```

Comment:
@rwaldron `opts.entryPoint` would be the entry point set for the cwd.  so it's ignoring at the relPath, not the resolved path of the actual file.  because of this, it was still ignoring the file in a nested dir when I was cd'd into that dir.